### ISSUE TITLE
HTTP HEAD responses include 'Content-Length: 0' header

### ISSUE THREAD

**Issue by Eyal-R** [medusa-project/cantaloupe#182]
Hello,

I generated a HTTP HEAD Request on a local Cantaloupe installation, I got:
```
HTTP/1.1 200 OK
Date: Sun, 10 Dec 2017 15:34:47 GMT
Content-Type: image/jpeg
Content-Disposition: inline
Cache-Control: max-age=2592000, public, no-transform
Date: Sun, 10 Dec 2017 15:34:47 GMT
Accept-Ranges: bytes
Server: Restlet-Framework/2.3.8
Vary: Accept-Charset, Accept-Encoding, Accept-Language, Accept
X-Powered-By: Cantaloupe/3.3.1
Link: <...>;rel="canonical"
Content-Length: 0
```
The content-length is 0, although the actual image returned in GET request of the same URL has more than bytes.

When I tried to ask for HTTP OPTIONS - I got HTTP 405 - Method not Allowed..

**Questions:**
1) Is HEAD supported?
2) For OPTIONS - it should indicate at least GET as an allowed HTTP action, don't you think?

**Comment by adolski** [comment]
Hi @Eyal-R,

Cantaloupe can't supply a `Content-Length` value because the internal image writers write directly to the response body, and the length is unknown at the time the headers are sent. I'm not sure why `Content-Length: 0` is appearing in a `HEAD` response -- it's probably a Restlet thing that I'll have to look into. But, yes, `HEAD` is available but `Content-Length` shouldn't be present.

As for HTTP OPTIONS, `405` is technically the correct response (for now) since `OPTIONS` is indeed not allowed. But, I agree that it should be and I'll create a separate ticket for that.

**Comment by Eyal-R** [comment]
Thanks! :)

**Comment by adolski** [comment]
https://github.com/restlet/restlet-framework-java/issues/1301

**Comment by adolski** [comment]
Update: probably a Jetty bug: https://github.com/eclipse/jetty.project/issues/596

**Comment by Eyal-R** [comment]
Thanks! but the issue you linked to looks closed..

**Comment by adolski** [comment]
I recently had to remove the Restlet framework for other reasons, and it looks like that has inadvertently fixed this. Jetty will now sometimes send a `Content-Length` header for small responses, but I haven't been able to get it to send `Content-Length: 0`. So, I'm going to close this, and the fix will appear in 4.1.
